src/string.ts

```diff
@@ -12,4 +12,8 @@ export const capitalize = <S extends string>(s: S): Capitalize<S> =>
 export const uncapitalize = <S extends string>(s: S): Uncapitalize<S> =>
   (S.isEmpty(s) ? s : s[0].toLowerCase() + s.slice(1)) as Uncapitalize<S>
 
+export const prefix = (prefix: string) => (s: string) => `${prefix}${s}`
+
+export const suffix = (suffix: string) => (s: string) => `${s}${suffix}`
```

Comment:
May I suggest using `<S extends string>` to return literal types instead of strings?

Reply:
I think there are a couple of solutions.

- Template literal as return type: 
```ts
declare const suffix: <A extends string, B extends string>(
  string: A,
) => (suffix: B) => `${A}${B}`

```
- Using the same signature above with no return type specified. The return type will be `string`.

I'd go for the first solution.

Any other idea?

Reply:
Why declaring the `B` generic in the first function, if it's used in the second one? That kills inference.

Reply:
That's right, moving the declaration of `B` in the second function fixes the problem of inference.